### DISCUSSION TITLE
Commemoratio layout problem

### DISCUSSION THREAD

**Discussion by Michaelskapel** [DivinumOfficium/divinum-officium#3988] [Q&A]
When I started the Nederlands version, I used the Latin version as my template and simply changed the language. I now see a different layout in the Missa on 12-04-2024, for instance, between the Nederlands (and also English) version and the Latin one. I noticed that in the mean time the Commemoratios have been removed from the Latin 12-04.txt file. Both the 12-04.txt file and the 12-04o.txt file were present when I started, and I still have the Commemoratios in the Nederlands 12-04.txt file. The (superfluous?) presence of the Commemoratios in the .txt file may be the reason of the different lay out. How shall we behave? Is it better to remove the Commemoratios from a .txt file if an o.txt file also exists?

**Comment by APMarcello3** [comment]
50c2042476

**Reply by Michaelskapel** [reply]
Is this a 'yes' to my more general question?

**Reply by APMarcello3** [reply]
Yes, correct.

**Comment by FAJ-Munich** [comment]
Maybe to clarify similar situations for the future, the question at hand has nothing to do about the existence of the `o.txt` file per se, but the particular situation that in the 1570 version, S. Barbara has not yet been superseded. Therefore, the Kalendarium for the 1960 version for December 4th reads:

```
12-04=12-04~12-04o=S. Petri Chrysologi Episcopi Confessoris Ecclesiae Doctoris=3=S. Barbarae Virginis et Martyris=1=
```

That is, the Oratio routine adds a commemoration to the feast from `12-04o` to the one in `12-04`. Having the `[Commemoratio Oratio]` in the latter file, creates a duplication. In other cases, the commemorated feast could be from a `cc.txt` or some other file depending on the history of a day in the Kalendarium. You can always double check in the Tabulae folder for reference.

**Reply by Michaelskapel** [reply]
Thanks! Actually, I have never paid attention to the Kalendarium. I assumed that the Latin version, which I used as a template, was okay and that I did not have to bother about the Kalendarium. But now it seems that sometimes corrections to the Latin version are necessary and that a similar corrections should actually be made in other versions, as well. Is there an easy way to get notified when this happens, so that others can update their version accordingly?

**Reply by FAJ-Munich** [reply]
Ideally, corrections to the Latin are also taken care of in all vernaculars by those correcting. Now, I have to admit that I have also forgotten sometimes that the Kalendarium is affecting both "horas" and "missa" and consequently ended up committing unwanted side effects to the Missa. 
I am working on some Perl tools to detect and avoid such things in the future. For the time being, these are pre-beta stage unfortunately.
